examples/src/walkable-path.ts

```diff
@@ -0,0 +1,374 @@
+  };
+}
+
+function updateQueryGUI(parent: GUI, data: TransitPathData): GUI {
+  const queryFolder = parent.addFolder('WalkablePath');
+  queryFolder.open();
+  queryFolder.add(data, 'modelId');
+  const updateItems = () => {
+    queryFolder.removeFolder(itemsFolder);
+    itemsFolder = updateItemsGUI(queryFolder, data.items, updateItems);
+  };
+  let itemsFolder = updateItemsGUI(queryFolder, data.items, updateItems);
+  return queryFolder;
+}
+
+function updateItemsGUI(parent: GUI, items: TransitPathItem[], onItemsUpdated: () => void) {
+  const itemsFolder = parent.addFolder('items');
+  itemsFolder.open();
+  itemsFolder.add(
+    {
+      addItem: () => {
+        const newItem = defaultPathItem();
+        items.push(newItem);
+        onItemsUpdated();
+      }
+    },
+    'addItem'
+  );
+  let index = 0;
+  for (const item of items) {
+    const itemIndex = index;
+    const indexFolder = itemsFolder.addFolder(`${index}`);
+    indexFolder.open();
+    createItemGUI(indexFolder, item, () => {
+      items.splice(itemIndex, 1);
+      onItemsUpdated();
+    });
+    index += 1;
+  }
+  return itemsFolder;
+}
+
+function createItemGUI(parent: GUI, item: TransitPathItem, onDelete: () => void) {
+  parent.add(
+    {
+      delete: onDelete
+    },
+    'delete'
+  );
+  const from = parent.addFolder('from');
+  from.open();
+  createWaypointGUI(from, item.from);
+  const to = parent.addFolder('to');
+  to.open();
+  createWaypointGUI(to, item.to);
+
+  const movingObjectSize = parent.addFolder('movingObjectSize');
+  movingObjectSize.add(item.movingObjectSize, 'diameter');
+  movingObjectSize.add(item.movingObjectSize, 'height');
+}
+
+function getWaypointData(waypoint: Waypoint) {
+  switch (waypoint.type) {
+    case Type.coordinates:
+      return waypoint.coordinates;
+    case Type.nodeId:
+      return waypoint.nodeId;
+    default:
+      throw new TypeError('');
+  }
+}
+
+function createWaypointGUI(parent: GUI, waypoint: Waypoint) {
+  const type = parent.add(waypoint, 'type', Object.values(Type));
+  const data = getWaypointData(waypoint);
+  let items = addToGui(parent, data, Object.keys(data));
+  type.onChange(() => {
+    for (const item of items) {
+      parent.remove(item);
+    }
+    const newData = getWaypointData(waypoint);
+    items = addToGui(parent, newData, Object.keys(newData));
+  });
+}
+
+function addToGui<T>(parent: GUI, target: T, items: string[]) {
+  const addedItems: GUIController[] = [];
+  for (const item of items) {
+    addedItems.push(parent.add(target, item));
+  }
+  return addedItems;
+}
+
+// Cognite Client, returns a function that queries the api for TransitPathResponse.
+interface TransitPathResponse {
+  items: [
+    {
+      segments: [
+        {
+          path: Coordinates[];
+        }
+      ];
+    }
+  ];
+}
+
+function createWalkablePathClient(project: string) {
+  const client: CogniteClient = new CogniteClient({ appId: 'Reveal Examples - WalkablePath' });
```

Comment:
Instead of creating a client, this should accept  a client as argument. This is to avoid multiple SDK clients.

Reply:
Changed function name parameters to take client